forge-game/src/main/java/forge/game/phase/PhaseHandler.java

```diff
@@ -1239,6 +1239,7 @@ public void endCombat() {
         game.getEndOfCombat().executeUntil();
         if (combat != null) {
             combat.endCombat();
+            game.getEndOfCombat().executeUntil(playerTurn);
```

Comment:
I had to hide this in here, or it would be killed right away by line 414 – I'm not sure this is the best way to do it